File: frontend/src/types/article.ts
export interface Article {
  id: number
  title: string
  content: string
  published: boolean
  authorId?: number
  coverUrl?: string | null
  /** 可选：关联项目 */
  relatedProjectId?: number | null
  /** 关联 AI 工具 slug */
  relatedToolSlugs?: string[]
  createdAt: string
  updatedAt: string
  deletedAt?: string | null
}

export interface ArticleCreateRequest {
  title: string
  content: string
  published?: boolean
  relatedProjectId?: number | null
  relatedToolSlugs?: string[]
}

export interface ArticleUpdateRequest {
  title: string
  content: string
  published: boolean
  relatedProjectId?: number | null
  relatedToolSlugs?: string[]
}

File: backend/src/main/java/com/zzh/personal_hub/blog/entity/Article.java
package com.zzh.personal_hub.blog.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "article")
public class Article {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    /** true = 已发布，可前台展示；false = 草稿/下架 */
    @Column(nullable = false)
    private Boolean published = false;

    /** 作者用户 id，对应 users.id */
    @Column(name = "author_id", nullable = false)
    private Long authorId;

    @Column(nullable = false, updatable = false)
    private Instant createdAt = Instant.now();

    @Column(nullable = false)
    private Instant updatedAt = Instant.now();

    @Column(name = "deleted_at")
    private Instant deletedAt; // null = 未删除
}